api/app/fireweather_bot/hourly_actuals.py

```diff
@@ -63,58 +102,37 @@ def _get_end_date(self) -> int:
-        data_df.rename(columns={'display_name': 'station_code'}, inplace=True)
-
-        # write to database using _session's engine
-        with app.db.database.get_write_session_scope() as session:
-            # write the data_df to the database one row at a time, so that if data_df contains >=1 rows that
-            # are duplicates of what is already in the db, the write won't fail for the unique rows
-            # NOTE: iterating over the data_df one Series (row) at a time is ugly, but until pandas is
-            # updated with a fix, this is the easiest work-around.
-            # See https://github.com/pandas-dev/pandas/issues/15988
-            # pylint: disable=unused-variable
-            for index, row in data_df.iterrows():
-                try:
-                    # Go from pandas to a python dict.
-                    data = row.to_dict()
-                    # Format and fix timestamp, make it be tz aware.
-                    data['weather_date'] = _fix_datetime(data['weather_date'])
-                    # Throw the data into the model, and persist in the database
-                    save_hourly_actual(session, HourlyActual(**data))
-                except IntegrityError:
-                    logger.info('Skipping duplicate record for %s @ %s',
-                                data['station_code'], data['weather_date'])
-                    session.rollback()
-
-    def construct_request_body(self):
-        return {
-            'Start_Date': self._get_start_date(),
-            'End_Date': self._get_end_date(),
-            'Format': 'CSV',
-            'cboFilters': config.get('BC_FIRE_WEATHER_FILTER_ID'),
-            'rdoReport': 'OSBH',
-        }
-
-
-def main():
-    """ Makes the appropriate method calls in order to submit a query to the BC FireWeather Phase 1 API
-    to get hourly values for all weather stations, downloads the resulting CSV file, writes
-    the CSV file to the database, then deletes the local copy of the CSV file.
+    async def run_wfwx(self):
+        """ Entry point for running the bot """
+        with Session() as session:
+            # Authenticate with idir.
+            authenticate_session(session)
+
+            station_codes = get_station_names_to_codes()
+
+            start_date = self._get_start_date()
+            end_date = self._get_end_date()
+
+            station_hourly_readings = await wildfire_one.get_hourly_readings(
+                station_codes, start_date, end_date)
+
+            logger.info("Station hourly readings: %s", station_hourly_readings)
+
+            with app.db.database.get_write_session_scope() as session:
+                for station_hourly_reading in station_hourly_readings.result():
+                    for hourly_reading in station_hourly_reading.values:
+                        save_hourly_actual(session, parse_hourly_actual(
+                            station_hourly_reading.station.code, hourly_reading))
+
+
+async def main():
```

Comment:
Main isn't awaited - so this script can't actually run right now. Having this be async also makes your unit tests more complicated than they need to be.

I think rather wrap the async bits in an event loop where needed and keep main normal.

Have a look in api/app/stations.py get_stations_synchronously - there's an example - basically looks like this:
```
loop = asyncio.new_event_loop()
asyncio.set_event_loop(loop)
the_result = loop.run_until_complete(run_my_async_thing_here)
```